pkg/proxy/iptables/proxier.go

```diff
@@ -1443,12 +1473,22 @@ func (proxier *Proxier) syncProxyRules() {
 	// Ref: https://github.com/kubernetes/kubernetes/issues/74839
 	// Ref: https://github.com/kubernetes/kubernetes/issues/117924
 	if !proxier.conntrackTCPLiberal {
-		proxier.filterRules.Write(
-			"-A", string(kubeForwardChain),
-			"-m", "conntrack",
-			"--ctstate", "INVALID",
-			"-j", "DROP",
-		)
+		if proxier.nfAcctCounters[metrics.IptablesCTInvalidDroppedPackets.Name] {
+			proxier.filterRules.Write(
+				"-A", string(kubeForwardChain),
+				"-m", "conntrack",
+				"--ctstate", "INVALID",
+				"-m", "nfacct", "--nfacct-name", metrics.IptablesCTInvalidDroppedPackets.Name,
+				"-j", "DROP",
+			)
+		} else {
+			proxier.filterRules.Write(
+				"-A", string(kubeForwardChain),
+				"-m", "conntrack",
+				"--ctstate", "INVALID",
+				"-j", "DROP",
+			)
+		}
```

Comment:
you can do

```
var nfacctRule []string
if proxier.nfAcctCounters[metrics.IptablesCTInvalidDroppedPackets.Name] {
        nfacctRule = []string{"-m", "nfacct", "--nfacct-name", metrics.IptablesCTInvalidDroppedPackets.Name}
}
proxier.filterRules.Write(
	"-A", string(kubeForwardChain),
	"-m", "conntrack",
	"--ctstate", "INVALID",
	nfacctRule,
	"-j", "DROP",
)
```

Reply:
This adds an extra white space if `nfacctRule` is empty, and messes the unit-tests.